### PR TITLE
ADMINUI-2315 Make MAC addresses searchable in AdminUI

### PR DESCRIPTION
**Pull request by Arc311**

### CODE REVIEW

File: lib/vms.js [hunk 1]
```diff
@@ -121,6 +121,24 @@ function list(req, res, next) {
                 } ]));
         }
         return listVmsWithIP(ip);
+    } else if (params.mac) {
+        req.sdc[req.dc].napi.getNic(params.mac, function (error, nic) {
+            if (error) {
+                return next(error);
+            }
+            var vmUuid = nic.belongs_to_uuid;
```
Comment (andyYatskov) [inline lib/vms.js]: you should check if vmUuid exist

File: lib/vms.js [hunk 2]
```diff
@@ -121,6 +121,24 @@ function list(req, res, next) {
...
+            req.sdc[req.dc].vmapi.getVm({uuid: vmUuid}, function (error, vm) {
+                if (error) {
+                    return next(error);
+                }
+                var vms = [];
+                if (vm) {
+                    vms.push(vm);
+                }
+                res.send(vms);
```
Comment (andyYatskov) [inline lib/vms.js]: res.send([vm]);

**Comment by andyYatskov** [comment]
🔢

File: lib/vms.js [hunk 3]
```diff
@@ -121,6 +121,23 @@ function list(req, res, next) {
...
+                return next(error);
+            }
+            var vmUuid = nic.belongs_to_uuid;
+            if (vmUuid) {
+                return req.sdc[req.dc].vmapi.getVm({uuid: vmUuid}, function (error, vm) {
+                    if (error) {
+                        return next(error);
+                    }
+                    res.send([vm]);
+                    next();
```
Comment (dan-luk) [inline lib/vms.js]: return

File: lib/vms.js [hunk 4]
```diff
@@ -121,6 +121,23 @@ function list(req, res, next) {
...
+                });
+            }
+            next();
```
Comment (dan-luk) [inline lib/vms.js]: return

btw what happens here?

**Comment by dan-luk** [comment]
🔢

**Comment by dan-luk** [comment]
🔢

**Comment by askfongjojo** [comment]
Merged as be2822b. Applied some refactoring from @johananlai to make the mac search code block consistent with ip search, with upfront mac address validation. I also added a check on 'belongs_to_type' so that it doesn't attempt to invoke vmapi when the mac belongs to a physical server.

**Pull request closed without merging**